src/Hedgehog/Gen.fs

```diff
@@ -247,26 +248,29 @@ module Gen =
     /// More or less the same logic as suchThatMaybe from QuickCheck, except
     /// modified to ensure that the shrinks also obey the predicate.
     let private tryFilterRandom (p : 'a -> bool) (r0 : Random<Tree<'a>>) : Random<Option<Tree<'a>>> =
-        let rec tryN k = function
+        let rec tryN (size1 : Size) (size2 : Size) : Random<Tree<'a> option> =
+            match Size.current size2 with
             | 0 ->
                 Random.constant None
             | n ->
-                let r = Random.resize (2 * k + n) r0
```

Comment:
If someone can explain to me what this is supposed to accomplish, I'd be very happy to hear it. I had no good name for this operation when moving it to the `Size` module.